src/components/Details/DetailedTrafficList.tsx

```diff
@@ -0,0 +1,261 @@
+import { Icon } from 'patternfly-react';
+import { TableGrid } from 'patternfly-react-extensions';
+import * as React from 'react';
+import { NodeType } from '../../types/Graph';
+import { REQUESTS_THRESHOLDS } from '../../types/Health';
+import { Direction } from '../../types/MetricsOptions';
+
+type DetailedTrafficProps = {
+  direction: Direction;
+  traffic: TrafficItem[];
+};
+
+export interface TrafficDataHttp {
+  protocol: 'http';
+  rates: {
+    http: string;
+    httpPercentErr?: string;
+  };
+}
+
+export interface TrafficDataGrpc {
+  protocol: 'grpc';
+  rates: {
+    grpc: string;
+    grpcPercentErr?: string;
+  };
+}
+
+export interface TrafficDataTcp {
+  protocol: 'tcp';
+  rates: {
+    tcp: string;
+  };
+}
+
+export type TrafficData = TrafficDataHttp | TrafficDataTcp | TrafficDataGrpc;
+
+export interface AppNode {
+  id: string;
+  type: NodeType.APP;
+  namespace: string;
+  name: string;
+  version: string;
+}
+
+export interface WorkloadNode {
+  id: string;
+  type: NodeType.WORKLOAD;
+  namespace: string;
+  name: string;
+}
+
+export interface ServiceNode {
+  id: string;
+  type: NodeType.SERVICE;
+  namespace: string;
+  name: string;
+}
+
+export interface UnknownNode {
```

Comment:
what's "unknown node" about? (I mean, what could it be for instance?)

Reply:
The "Unknown" node type is to deal with the special "unknown" node in the graph.  I'm not sure it's actually relevant to the detail pages as it's not a real app/service/workload.

Reply:
This is to, at least, show it as a source or a destination of traffic (see the "inbound" in the image of the main comment). I can remove this and that row won't be shown. I don't know if it is better to show or not. So... you say.

Reply:
It should not be a destination, but I guess as a source it is valid.